File: Excel/800-899/834/834 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/800-899/834/834 Group Rowwise.xlsx"
input = read_excel(path, range = "A2:F10")
test  = read_excel(path, range = "H2:I6")

result = input %>%
  as.matrix() %>%
  apply(1, function(x) {
    if (which(input == x[1], arr.ind = TRUE)[1, "row"] %% 2 == 0) {
      paste(x %>% na.omit(), collapse = "+")
    } else {
      paste(x %>% na.omit(), collapse = "")
    }
  }) %>%
  matrix(ncol = 2, byrow = TRUE) %>%
  as_tibble() %>%
  mutate(V2 = map_int(V2, ~ eval(parse(text = .x)))) %>%
  set_names(c("Group", "Sum"))

all.equal(result, test)
# [1] TRUE

File: Excel/800-899/834/834 Challenge.py
import pandas as pd
import numpy as np

path = "800-899/834/834 Group Rowwise.xlsx"
input = pd.read_excel(path, usecols="A:F", skiprows=1, nrows=9)
test = pd.read_excel(path, usecols="H:I", skiprows=1, nrows=4)
input_matrix = input.to_numpy()
result = pd.DataFrame({
    "Group": ["".join([str(x) for x in row if pd.notna(x)]) for row in input_matrix[::2]],
    "Sum": [eval("+".join([str(x) for x in row if pd.notna(x)])) for row in input_matrix[1::2]]
})

print(result.equals(test))
# True